### PR TITLE
parse command usage and flagset usage from code

### PR DESCRIPTION
**Pull request by micbar**
## Description
- Adds automated parsing of command usage and flagset usage and defaults from code to docs
- Add it to `make docs`
- Will be auto generated during CI run

## Deps
- uses   local replace in go.mod
- needs PR https://github.com/owncloud/flaex/pull/3 to be merged first

## Test it
```console
make docs
cd hugo
hugo -D server
````
go to http://localhost:1313/

Screenshot_2020-04-10 Configuration ownCloud

### CODE REVIEW

**Comment by micbar** [comment]
@butonic @kulmann Something for the docs initiative 😄 💪

**Comment by micbar** [comment]
@xoxys the `TOC` is limited to 2 levels. Is that intentional?

**Comment by xoxys** [comment]
@micbar see: https://github.com/owncloud/owncloud.github.io/blob/source/config.yaml#L16

**Comment by micbar** [comment]
I tried changing it, no effect.

**Comment by xoxys** [comment]
> I tried changing it, no effect.

@micbar I don't know how to reproduce it ... Your file (e.g. configuration.md) simply has only two heading levels. I added a third level manually and it is working as intended:

grafik

**Comment by xoxys** [comment]
`Getting started` section which has three levels also works for me

**Comment by micbar** [comment]
ok, yes you are right about the levels, the current pr has only 3 levels.
I tried it with 4.

Let me check again.

**Comment by xoxys** [comment]
Or maybe you could explain what you try to achieve

**Comment by micbar** [comment]
@xoxys Thanks for debugging.

This TOC feature is only working until hugo `v0.59.0`

File: docs/getting-started.md [hunk 1]
```diff
@@ -1,27 +1,64 @@
...
+- Running ocis currently needs a working Redis caching server
+- The default promary storage location is `/var/tmp/reva/data`. You can change that value by configuration.
+
+## Usage
 
-TBD
+The program provides a few sub-commands on execution. The available configuration methods have already been mentioned above. Generally you can always see a formated help output if you execute the binary via `ocis --help`.
 
-#### Binaries
+### Server
```
Comment (kulmann) [inline docs/getting-started.md:50]: ocis-reva has no server subcommand

Comment (micbar) [reply docs/getting-started.md]: copy & pasta :-)

**Pull request merged**